src/libs/actions/Workflow.ts

```diff
@@ -80,11 +78,6 @@ function createApprovalWorkflow(policyID: string, approvalWorkflow: ApprovalWork
     ];
 
     const failureData: OnyxUpdate[] = [
-        {
-            onyxMethod: Onyx.METHOD.MERGE,
-            key: ONYXKEYS.APPROVAL_WORKFLOW,
-            value: {...approvalWorkflow, isLoading: false},
-        },
```

Comment:
Nope. It would create the same bug as the current issue but on the failing of the API request instead of the current issue that is happening on success.

Reply:
Why would it create the same bug?

Reply:
We are adding the info back so the user can fix the mistakes.. Speaking of mistakes, do you know if this API can return errors and where are they being displayed as currently it does not seem we can access that workflow page without overwriting its data first

Reply:
@s77rt we always load the draft with the current workflow when we open an edit workflow page (we need to save the draft approval workflow because we have different sub-pages in the workflow edit flow) so let's say the user is editing a different workflow when the previous save API fails so you are suggesting to revert the previous draft workflow which is wrong.

Reply:
I see what you are saying but what if the API request failed? How is the user supposed to know that it failed? Or is this not supposed to happen?

Reply:
On failure the workflow changes the user made will be reverted
https://github.com/Expensify/App/blob/1240b87a77ffebd9b1fb6b8bd42450fd91491c12/src/libs/actions/Workflow.ts#L165-L171
but approvalWorkflow, as I clearly indicated in my proposal, is the onyx key we use to save temporary draft workflow that we are editing and we don't need to revert that onFailure. Is it clear now?